src/main/java/com/google/devtools/build/lib/worker/WorkerSpawnRunner.java

```diff
@@ -109,12 +110,17 @@ public WorkerSpawnRunner(
 
   @Override
   public String getName() {
-    return "worker";
+    if (this.supportsMultiplexWorkers) {
+      return "multiplex-worker";
+    } else {
+      return "worker";
+    }
   }
 
   @Override
   public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
       throws ExecException, IOException, InterruptedException {
+    this.supportsMultiplexWorkers = Spawns.supportsMultiplexWorkers(spawn);
```

Comment:
It looks like a bad idea. Can someone help?

Reply:
Yes, I don't think this will work (reliably), because the WorkerSpawnRunner instance is shared for all worker strategy executions (multiplex and non-multiplex). I think this might cause the name of all running worker actions to change to "multiplex-worker" or "worker", depending on whether the most recently executed action was a multiplexed one or not.

I can't immediately think of a good solution for this, except maybe instantiating two WorkerSpawnRunners (one for multiplexed workers, one for non-multiplexed workers). But it might make the code much more complex.

Maybe @susinmotion or @jmmv have a good idea.

Reply:
What about appending 'multiplex' to the existing name of the worker if it's multiplex?

Reply:
I think the challenge is that when `WorkerSpawnRunner` is constructed, we don't have `Spawn` object to know if it's multiplex or not. I don't think we should set it in `exec()` since `this.supportsMultiplexWorkers` could be undefined at a certain period of time.

@susinmotion do you mean we could undo the change in `getName()`? It's nice to see information when build is in progress though, like
```
<Mnemonic> <Target> 10s multiplex-worker
```

Reply:
Yes, I did mean undoing the change in getName(), but I also see why that'd be nice info to have.
Let me take a closer look!

Reply:
What we care about is the name that is used for logging, right?

If so, why not just change the `context.report()` calls in this file to specify the exec-specific name instead of calling `getName()`? They all happen within `exec()`, so at that point we have the right context information.

Reply:
Sounds like a great idea. Just force-pushed, please take another look.